plugins/social_network/social_network.cpp

```diff
@@ -282,6 +282,96 @@ namespace golos { namespace plugins { namespace social_network {
+                    cr.author_golos_payout_value += op.steem_payout;
+                    cr.author_gests_payout_value += op.vesting_payout;
+                });
+            }
+        }
+
+        result_type operator()(const comment_payout_update_operation& op) const {
+            if (!db.has_index<comment_reward_index>())
+                return;
+
+            const auto &comment = db.get_comment(op.author, op.permlink);
+
+            const auto& cr_idx = db.get_index<comment_reward_index>().indices().get<golos::plugins::social_network::by_comment>();
+            auto cr_itr = cr_idx.find(comment_id_type(comment.id));
+            if (cr_itr == cr_idx.end()) {
+                db.create<comment_reward_object>([&](golos::plugins::social_network::comment_reward_object& cr) {
+                    cr.comment = comment.id;
+                    cr.total_payout_value = op.total_payout_value;
+                    cr.beneficiary_payout_value = op.beneficiary_payout_value;
+                    cr.curator_payout_value = op.curator_payout_value;
+                });
+            } else {
+                db.modify(*cr_itr, [&](comment_reward_object& cr) {
+                    cr.total_payout_value += op.total_payout_value;
+                    cr.beneficiary_payout_value += op.beneficiary_payout_value;
+                    cr.curator_payout_value += op.curator_payout_value;
+                });
+            }
+        }
+
+        result_type operator()(const curation_reward_operation& op) const {
+            if (!db.has_index<comment_reward_index>())
+                return;
+
+            const auto &comment = db.get_comment(op.comment_author, op.comment_permlink);
+
+            const auto& cr_idx = db.get_index<comment_reward_index>().indices().get<golos::plugins::social_network::by_comment>();
+            auto cr_itr = cr_idx.find(comment_id_type(comment.id));
+            if (cr_itr == cr_idx.end()) {
+                db.create<comment_reward_object>([&](golos::plugins::social_network::comment_reward_object& cr) {
+                    cr.comment = comment.id;
+                    cr.curator_gests_payout_value = op.reward;
+                });
+            } else {
+                db.modify(*cr_itr, [&](comment_reward_object& cr) {
+                    cr.curator_gests_payout_value += op.reward;
+                });
+            }
+        }
+
+        result_type operator()(const comment_benefactor_reward_operation& op) const {
+            if (!db.has_index<comment_reward_index>())
+                return;
+
+            const auto &comment = db.get_comment(op.author, op.permlink);
+
+            const auto& cr_idx = db.get_index<comment_reward_index>().indices().get<golos::plugins::social_network::by_comment>();
+            auto cr_itr = cr_idx.find(comment_id_type(comment.id));
+            if (cr_itr == cr_idx.end()) {
+                db.create<comment_reward_object>([&](golos::plugins::social_network::comment_reward_object& cr) {
```

Comment:
Rewards come here in order of send. So you can accumulate counters and update only on comment_payout_update_operation